pwa/context/ToastContext.tsx

```diff
@@ -25,12 +25,15 @@ export const ToastProvider: React.FC<React.PropsWithChildren> = ({ children }) =
 
   return (
     <ToastContext.Provider
-      value={(toast) => {
-        setToastList((prevToasts) => [...prevToasts, toast]);
-        setTimeout(() => {
-          setToastList((prevToasts) => [...prevToasts.slice(1)]);
-        }, 3000);
-      }}
+      value={useCallback(
+        (toast) => {
+          setToastList((prevToasts) => [...prevToasts, toast]);
+          setTimeout(() => {
+            setToastList((prevToasts) => [...prevToasts.slice(1)]);
+          }, 3000);
+        },
```

Comment:
Don't pass a callback directly and generally don't use directly hooks returned values